### ISSUE TITLE
Example of ground_truth_dict and result_dict

### ISSUE THREAD

**Issue by Ybakman** [LeMuecke/mapcalc#1]
I couldn't understand how to initialize ground_truth_dict and result_dict dictionaries when we have multiple images and predictions. Can you give an example? Thank you.

**Comment by klimaschkas** [comment]
**ground_truth_dict** looks like the following:

```
<class 'dict'>: {
       'boxes': array([[ 871.,  572.,  925.,  664.],
       [ 948.,  592., 1010.,  684.],
       [ 874.,  705.,  941.,  815.],
       ...
       [ 855.,   85.,  867.,  103.],
       [ 850.,  125.,  860.,  152.],
       [ 849.,  122.,  860.,  144.],
       [ 849.,  119.,  860.,  141.]], dtype=float32), 

       'labels': array([ 4,  4,  4, 10,  2, 10,  1,  1,  8,  2,  8, 10, 10,  7, 10,  7, 10,
       10, 10, 10, 10,  8, 10,  8,  8,  8, 10, 10, 10, 10, 10,  8, 10,  8,
       ...
       1, 10,  2,  2,  4,  4,  1,  1, 10,  2,  2, 10,  2, 10,  2, 10,  2,
       8,  2, 10,  2,  8,  2, 10,  2,  2], dtype=int64)}

```

**result_dict** looks like the following:

```
<class 'dict'>: {
       'boxes': array([[ 873.,  571.,  923.,  666.],
       [ 945.,  588., 1011.,  685.],
       [ 882.,  700.,  943.,  817.],
       ...
       [ 857.,  328.,  883.,  365.],
       [ 581.,  763.,  615.,  804.],
       [ 865.,  400.,  905.,  439.]]), 

       'labels': array([ 4,  4,  4,  4,  4,  4,  5, 10, 10, 10,  4, 10, 10, 10,  1,  5, 10,
       10,  2, 10, 10,  4,  2, 10, 10, 10,  4, 10,  2,  1, 10, 10,  2, 10,
       ...
       10, 10,  1, 10, 10,  2, 10,  4,  2,  2,  1, 10, 10, 10,  2],
      dtype=int64), 

      'scores': array([0.99056727, 0.98965424, 0.93990153, 0.9157755 , 0.8776852 ,
       0.8343509 , 0.7326441 , 0.6726452 , 0.65791756, 0.6477029 ,
       ...
       0.2758158 , 0.2751832 , 0.27424183, 0.27214053, 0.27162442,
       0.26996344, 0.26646382, 0.2605154 , 0.26025057, 0.25834218],
      dtype=float32)}
```

So, in ground_truth_dict, boxes contains an array of arrays with 4 floats, in this case the left corner of the bounding box, the bottom corner of the bounding box, the right corner and the top corner. Labels are integers with every integer representing one class.
In result_dict, boxes has the same format as in ground_truth_dict, labels also as in ground_truth_dict and scores contains a float about how sure the network was about the prediction.

Hope that helps, I meant to include more detailed explanation and examples but simply forgot. If you still struggle, just reply in this issue.

**Comment by sheldonsebastian** [comment]
`    if not len(ground_truth_dict['boxes']) == len(ground_truth_dict['labels']) == len(result_dict['boxes']) == len(
            result_dict['labels']) == len(result_dict['scores']):
        raise ValueError("The arrays in the dictionary have different sizes. They need to have the same size.")`

I am not understanding how to compute mAP for multiple images. The code says that the len(predicted_boxes) == len(ground_truth_boxes). But this will never be true, since object detection models predict more boxes than actual ground truth boxes.

**Comment by klimaschkas** [comment]
That is indeed a bug, when writing the code the use case was having a fixed amount of predictions, the remaining ones were dropped.
I'll enhance the model by the possibility to have a different amount of ground truth and predictions.

**Comment by klimaschkas** [comment]
@sheldonsebastian I just published v0.2.1 which removes the restrictions you mentioned. The number of predictions can now differ from the ground truth.
v0.2.0 had a flag that would allow stripping off the predictions that are too much, but I then realized that the mAP calculations actually automatically handles that. My bad, it's been some days since doing object recognition.
